### DISCUSSION TITLE
MSVC17

### DISCUSSION THREAD

**Discussion by Ethan-Bit-256** [raysan5/raylib#3173] [Q&A]
Is it possible for me to link raylib msvc16(64bit) to vs2022(msvc17) and have it work perfectly fine?

**Comment by raysan5** [comment]
@Ethan-Bit-256 I don't know, I haven't tried it but you can build raylib with VS2017.

**Comment by orcmid** [comment]
@Ethan-Bit-256 If you're talking about a pre-built DLL and/or LIB, it will depend on agreement of raylib version and, potentially, compiler options.

You can obtain very clean static linking with VS2022 (VC <strike>17.6.0</strike>17.6.5 currently) if you use the latest raylib stable source (4.5) and compile a cache of pre-built raylib components for linking into your code.  This will generally assure compatibility between raylib code and the raylib headers you use from your code, as well as ensuring consistent compiler options.  

There's a fixture for that in `raylib\projects\scripts\build-windows.bat`.  You can use it from the `x64 Native Tools Command Prompt for VS 2022`.  

I made a derivative, VCrayApp, that automates things a bit differently.  A pretty-stable 0.1.0 version is available at (https://orcmid.github.io/nfoTools/dev/D211101/).   The documentation has many gaps still.   I welcome questions and suggestions.
